From d902d31fd48ed1b9cb4bad1b2cc8cb159cd4985d Mon Sep 17 00:00:00 2001
From: Janne Pulkkinen <janne.pulkkinen@protonmail.com>
Date: Wed, 23 Nov 2022 19:23:17 +0200
Subject: [PATCH 1/2] Disable "64-bit prefix" warning popup

The 64-bit prefix is used with every prefix on Proton, so there is no
need to show the popup every time.
---
 src/winetricks | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/winetricks b/src/winetricks
index c0c3ff9..76a7007 100755
--- a/src/winetricks
+++ b/src/winetricks
@@ -5313,11 +5313,17 @@ winetricks_set_wineprefix()
         W_COMMONFILES="$(w_pathconv -u "${W_COMMONFILES_WIN}")"
 
         # 64-bit prefixes still have plenty of issues:
+        # Protontricks patch: Don't display a warning popup for the 64-bit
+        # prefix
+        _winetricks_gui="${WINETRICKS_GUI}"
+        WINETRICKS_GUI=none
         case ${LANG} in
             ru*) w_warn "Вы используете 64-битный WINEPREFIX. Важно: многие ветки устанавливают только 32-битные версии пакетов. Если у вас возникли проблемы, пожалуйста, проверьте еще раз на чистом 32-битном WINEPREFIX до отправки отчета об ошибке." ;;
             pt*) w_warn "Você está usando um WINEPREFIX de 64-bit. Observe que muitos casos instalam apenas versões de pacotes de 32-bit. Se você encontrar problemas, teste novamente em um WINEPREFIX limpo de 32-bit antes de relatar um bug." ;;
             *) w_warn "You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug." ;;
         esac
+        WINETRICKS_GUI="${_winetricks_gui}"
+
     else
         WINE64="false"
         WINE_ARCH="${WINE}"
-- 
2.38.1

